pkg/continuousprofiling/profiler.go

```diff
@@ -0,0 +1,125 @@
+package continuousprofiling
+
+import (
+	"runtime"
+
+	"github.com/grafana/pyroscope-go"
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/logging"
+)
+
+var (
+	ErrApplicationName           = errors.New("the ApplicationName must be defined")
+	ErrServerAddress             = errors.New("the ServerAddress must be defined")
+	ErrAtLeastOneProfileIsNeeded = errors.New("at least one profile is needed")
+
+	log = logging.LoggerForModule()
+
+	DefaultProfiles = []pyroscope.ProfileType{
+		pyroscope.ProfileCPU,
+		pyroscope.ProfileAllocObjects,
+		pyroscope.ProfileAllocSpace,
+		pyroscope.ProfileInuseObjects,
+		pyroscope.ProfileInuseSpace,
+		pyroscope.ProfileGoroutines,
+		pyroscope.ProfileMutexCount,
+		pyroscope.ProfileMutexDuration,
+		pyroscope.ProfileBlockCount,
+		pyroscope.ProfileBlockDuration,
+	}
+)
+
+// DefaultConfig creates a new configuration with default properties.
+func DefaultConfig() *pyroscope.Config {
+	return &pyroscope.Config{
+		ApplicationName:   "AppName",
+		ServerAddress:     env.ContinuousProfilingServerAddress.Setting(),
+		BasicAuthUser:     env.ContinuousProfilingBasicAuthUser.Setting(),
+		BasicAuthPassword: env.ContinuousProfilingBasicAuthPassword.Setting(),
+		ProfileTypes:      DefaultProfiles,
+		Logger:            nil,
+	}
+}
+
+type OptionFunc func(*pyroscope.Config)
+
+// WithAppName sets the ApplicationName
+// Default: AppName
+func WithAppName(appName string) OptionFunc {
+	return func(cfg *pyroscope.Config) {
+		cfg.ApplicationName = appName
+		if env.ContinuousProfilingAppName.Setting() != "" {
+			// If ROX_CONTINUOUS_PROFILING_APP_NAME is set, we override the appName
+			cfg.ApplicationName = env.ContinuousProfilingAppName.Setting()
+		}
+	}
+}
```

Comment:
This is a subject to an overwrite from the env, so maybe we shall call it `WithDefaultAppName`?

Reply:
I've changed the logic a bit but your suggestion still makes sense. I've changed the name to `WithDefaultAppName` in the last commit.